api/src/main/java/org/openmrs/module/appointmentscheduling/api/db/hibernate/HibernateTimeSlotDAO.java

```diff
@@ -72,5 +77,55 @@ else if (fromDate != null && toDate != null && !fromDate.before(toDate)) {
 		return super.sessionFactory.getCurrentSession().createCriteria(TimeSlot.class)
 		        .add(Restrictions.eq("appointmentBlock", appointmentBlock)).add(Restrictions.eq("voided", false)).list();
 	}
+
+	@Override
+	public TimeSlot getResourceTimeslot(Date date, Location location) throws DAOException {

```

Comment:
Per my comment above, can you give a better description of what this method is intended to do?